### PR TITLE
closes #71 closes #74 closes #78 updates from admiralpeds experience

### LINKED ISSUE #71: Documentation: Package extension guidance should link to admiraldev site
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

@bms63 @manciniedoardo should https://pharmaverse.github.io/admiraltemplate/cran-release/index.html#package-extension-guidance be replaced with a link to https://pharmaverse.github.io/admiraldev/articles/package_extensions.html now? i trust the admiraldev site version is the one we should refer to for admiralpeds

### LINKED ISSUE #74: Update pkgdown to use dev mode
Update pkgdown to use dev mode

### LINKED ISSUE #78: Add data-raw folder
See here: https://github.com/pharmaverse/admiral/issues/2427

### PR DESCRIPTION
**Pull request by rossfarrugia**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by rossfarrugia** [comment]
@bms63 @manciniedoardo here's a bunch of updates i think would have helped make admiralpeds easier so hopefully passing on the favour for admiralmetabolic. shout if you disagree with any though.

one big update was the news file here where my justification is that people dont need to track the changes of this template repo as they always should only care about the latest version when they come to use it. we should set it up so that each pkg ext team has the minimum amount of updates and clarity on how they should update, rather than treating this template as a package where we have to maintain a changelog of changes.

File: _pkgdown.yml [hunk 1]
```diff
@@ -65,3 +70,15 @@ navbar:
       menu:
       - text: Creating ADXX
         href: articles/adxx.html
+    history:
+      icon: fa-history
+      href: articles/website-versions.html
+      aria-label: Previous Release Websites
+    slack:
+      icon: fa-slack
+      href: https://app.slack.com/client/T028PB489D3/C02M8KN8269
```
Comment (manciniedoardo) [inline _pkgdown.yml]: We now use this link for slack across the admiral packages: https://pharmaverse.slack.com/

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
LGTM, one minor comment about the slack link, agree on the NEWS.md front.

File: _pkgdown.yml (hunk 1 above)
Comment (rossfarrugia) [reply _pkgdown.yml]: good spot!

File: DESCRIPTION [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Type: Package
-Package: admiraltemplate
-Title: ADaM in R Asset Library - Extension
-Version: 0.0.3
+Package: `<<INSERT PACKAGE EXT NAME>>'
```
Comment (bms63) [inline DESCRIPTION:2]: This crashes our checks - maybe we could have a startup message that refers to a function that when calls updates everything for the extension pkg?  Maybe fun issue for someone to work on from community?

File: NEWS.md
Comment (bms63) [inline NEWS.md:1]: Oh I sort of like the news … history of what we have done!  Was this confusing for folks?  I like the message it sends that we catalog changes even for our template repo :) but if y’all think it’s got to go then all good

File: NEWS.md [hunk 3]
```diff
@@ -1,62 +1,8 @@
...
-## Documentation
-
- - Stuff that you did (#ISSUE-NUMBER)
-
-## Various
-
- - Stuff that you did (#ISSUE-NUMBER)
+- Initial package release focused on `<<ADD INITIAL SCOPE>>`
 
+Note that after first release you should then start maintaining this `NEWS.md` file for every user-facing update.
```
Comment (bms63) [inline NEWS.md:5]: This is good advice - what if this lives at the op and just says delete this and the old news?

**Review by bms63 — CHANGES_REQUESTED** [review]
Way cool Ross!  Just some thoughts on a few things.  I’m not super passionate about any of them.

Other ideas that maybe others can work on:

- snapshot tests example with two examples one works and one errors (this really makes writing tests so much less painful)
- Remotes in DESCRIPTION 

these overkill? I can make into issues if not

**Review by bms63 — COMMENTED** [review]
Can we remove renv reference in readme

File: DESCRIPTION (hunk 2 above)
Comment (rossfarrugia) [reply DESCRIPTION:2]: nice idea for the function (although in reality its questionable whether worth the investment given it'd only ever be needed once or twice a year likely?). for the CI i think fine for them to crash for admiraltemplate as its not a real package anyway

File: NEWS.md
Comment (rossfarrugia) [reply NEWS.md:1]: I'd say people can look to admiral or other pkg ext news files for examples. the confusing thing in having it was that every new pkg ext team starts logging every change from the start but in reality you only need this populated after you make first release, as everything until then is just release 1 features and not changes that users need to be aware of.

File: NEWS.md (hunk 3 above)
Comment (rossfarrugia) [reply NEWS.md:5]: I'd leave it as is, as they can just delete this section at the bottom and then they almost have their first NEWS.md ready to go with the top lines i gave here

**Comment by rossfarrugia** [comment]
@bms63 @manciniedoardo thanks for the review! replies added above, and renv already removed from README. if you're ok with my replies feel free to approve/merge or make changes as you wish if otherwise. for the other ideas then go ahead and make issues/PRs as needed. i just wanted to get these all in ahead of whenever admiralmetabolic repo gets created.

File: DESCRIPTION (hunk 2 above)
Comment (bms63) [reply DESCRIPTION:2]: Should we mention that once you fix this the ci will work again?

Comment (rossfarrugia) [reply DESCRIPTION:2]: not sure where would you put this note. it feels obvous for me as there are lots of places you need to update the placeholders, and the person first making an admiral pkg ext is never going to be a complete newbie. feel free to make a commit though if you think thats useful.

Comment (bms63) [reply DESCRIPTION:2]: I put it to the community to update if they want to - but agree low priority, but a fun one. https://github.com/pharmaverse/admiraltemplate/issues/80

File: NEWS.md
Comment (bms63) [reply NEWS.md:1]: okay that makes sense.  I was referring to the actual news for working on admiraltemplate, but it sounds like that is going to be jettison which is fine.

**Review by bms63 — APPROVED** [review]
